cli/src/main/kotlin/commands/ScannerCommand.kt

```diff
@@ -201,10 +229,22 @@ class ScannerCommand : CliktCommand(name = "scan", help = "Run external license
 
         val config = globalOptionsForSubcommands.config
 
-        val ortResult = if (experimental) {
-            runExperimental(config)
-        } else {
-            run(config)
+        val ortResult = when (scannerOption) {
```

Comment:
You could use `when (val scanner = scannerOption) {` here and then `scanner` instead of `scannerOption` below to make the smart-cast work (and probably inline the `scannerFactory` variable).